File: 序号/2609. 最长平衡子字符串.py
class Solution:
    def findTheLongestBalancedSubstring(self, s: str) -> int:
        c0, c1, ans = 0, 0, 0
        pre = "2"
        for x in s:
            if x == "0":
                if pre != "0":
                    c0 = 1
                else:
                    c0 += 1
            else:
                if pre != "1":
                    c1 = 1
                else:
                    c1 += 1
                ans = max(ans, min(c0, c1) * 2)
            pre = x
        return ans

File: 序号/2609. 最长平衡子字符串.cpp
class Solution {
public:
    int findTheLongestBalancedSubstring(string s) {
        size_t len=s.length();
        int a,b,t,ans=0;
        if(s[0]=='0'){
            a=1;b=0;
        }else{
            a=b=0;
        }
        for(size_t i=1;i<len;i++){
            char c = s[i];
            if(c=='0'){
                if(s[i-1]=='0'){
                    a+=1;
                }else{
                    t=min(a,b);
                    ans=max(ans,t);
                    a=1;b=0;
                }

            }else{
                b+=1;
            }
        }
        t=min(a,b);
        ans=max(ans,t);
        return ans*2;
    }
};